nexusutils/dataconverter/convert.py

```diff
@@ -106,12 +107,16 @@ def convert(input_file: Tuple[str],
         raise Exception("The chosen NXDL isn't supported by the selected reader.")
 
     if objects is not None:
-        data = data_reader().read(template=Template(template),  # type: ignore[operator]
-                                  file_paths=input_file,
-                                  objects=objects)
+        data = data_reader(**kwargs).read(  # type: ignore[operator]
+            template=Template(template),
+            file_paths=input_file,
+            objects=objects
+        )
     else:
-        data = data_reader().read(template=Template(template),  # type: ignore[operator]
-                                  file_paths=input_file)
+        data = data_reader(**kwargs).read(  # type: ignore[operator]
+            template=Template(template),
+            file_paths=input_file
+        )
```

Comment:
I'll check the rest later. I just wanted to check this out. 

So you're now passing some parmas to the init func as well. That's cool and nice that it's here now. But my question to you is, should we do the same with the call to the read func then? I thought what I did might be simplifying it for folks who write the readers. But of course this kwargs approach is better, if understood and used right. And it will get rid of this unnecessary if statement to check for `objects`.

Reply:
Yes, I think it would be nice to insert the **kwargs in the function, too. Then the readers could also define own parameter names, e.g. `key_value_map` or `raw_data`. If people don't really know how to deal with it we still can keep the objects in the example (which also should still just work if we pass `**kwargs` and remove the `objects` parameter from the `convert` functions parameters).

Reply:
Ah but we'd had to add **kwargs to each readers `read` function. Or do you have a nice approach in mind how to handle this in the base class exclusively?

Reply:
We could continue to pass it to the `__init__` and have a `read` wrapper which looks something like this
```python

def __init__(self, **kwargs):
    process_class_kwargs(**kwargs) # Not sure if **kwargs copies...
    del kwargs['processed_key'].   # But it's just some boilerplate for handling **kwargs here

   self._reduced_kwargs = kwargs

def read_wrapper(self, template, file_paths):
    return self.read(template, file_paths, **self._reduced_kwargs)
```

```process_class_kwargs``` just consumes the kwargs which should be processed at class initialisation. The rest is passed to read. We would then call `read_wrapper` in the `convert` function. We could probably also drop all kwargs which are not allowed for the parser and log a warning for that (we would need to get a list of params of the read functions... don't know if this is easily possible).

-- just a quick idea which would leave the writing of parsers and cli interfacing untouched.

Reply:
Something like this can work. But it will be confusing, implementation wise, for newer `_reader.py`'s. It's our own secret implementation of something so common. If we implement **kwargs, then updating the reader codes will definitely be better for documentation. What do you think? If you like, I can help with it. Or we can do it separate to this PR. Your call.

Reply:
I think it's good to have the **kwargs approach. Maybe it's better then to just pass the **kwargs to the read function and not to the `__init__` of the class, my solution was just thinking about how it could be if we pass it to both. In this case we wouldn't want to have the parameter in both, the init and the read function.

I can just do it. I think it won't be to hard to update. We can just quickly do it here, the merge is not too urgent.

Reply:
It's updated. We don't even have to change anything in the `read` funcs as objects will be passed automatically and is set to None in the base reader. The only change is that `objects` now has to be passed explicitly where before it could also be passed as `*args`. I think nobody ever used a non-keyworded objects argument, so it'll be fine.